tasks/configure.js

```diff
@@ -97,76 +106,62 @@ function task (opts) {
-						} else {
-							patch[key] = desired[key];
-						}
-					});
-
-					Object.keys(overrides).forEach(function (key) {
-						patch[key] = overrides[key];
-					});
-
-					Object.keys(patch).forEach(function (key) {
-						if (patch[key] === null) {
-							console.log('Deleting config var: ' + key); // eslint-disable-line no-console
-						} else if (patch[key] !== current[key]) {
-							console.log('Setting config var: ' + key); // eslint-disable-line no-console
-						}
-					});
-
-					console.log('Setting environment keys', Object.keys(patch)); // eslint-disable-line no-console
-
-					return fetch('https://api.heroku.com/apps/' + target + '/config-vars', {
-						headers: authorizedPostHeaders,
-						method: 'patch',
-						body: JSON.stringify(patch)
-					});
-				})
-				.then(response => {
-					if (response.status !== 200) return response.json().then(({id, message}) => Promise.reject(new Error(`Heroku Error - id: ${id}, message: ${message}`)));
-					console.log(target + ' config vars are set'); // eslint-disable-line no-console
-				}));
-			});
+	console.log('Retrieving pipeline details from Heroku...'); // eslint-disable-line no-console
+
+	// TODO: Investigate what herokuAuthToken is doing
+	const [ authToken, pipelineId ] = await Promise.all([
+		herokuAuthToken(),
+		getPipelineId(source)
+	]);
+
+	const herokuConfigVars = new HerokuConfigVars({ target, pipelineId, authToken });
+
+	const serviceData = await getServiceData(source, opts.registry);
+
+	console.log('Retrieving current and desired config vars...'); // eslint-disable-line no-console
+
+	const [ desired, current ] = await Promise.all([
+		fetchFromVault(serviceData),
+		herokuConfigVars.get()
+	]);
+
+	desired['SYSTEM_CODE'] = serviceData.code;
+
+	desired['___WARNING___'] = 'Don\'t edit config vars manually. Use the Vault UI.';
+	let patch = {};
+
+	Object.keys(current).forEach(function (key) {
+		if (!key.startsWith('HEROKU_') && !FORBIDDEN_ATTACHMENT_VARIABLES.includes(key)) {
+			patch[key] = null;
+		}
+	});
+
+	Object.keys(desired).forEach(key => {
+		if (FORBIDDEN_ATTACHMENT_VARIABLES.includes(key)) {
+			throw new Error(`\nCannot set environment variable '${key}' as this variable name is used for an attachment variable by Heroku, `
+				+ 'if this is for an external service, please use a different environment variable name in your app\n');
+		} else {
+			patch[key] = desired[key];
+		}
+	});
+
+	Object.keys(overrides).forEach(function (key) {
+		patch[key] = overrides[key];
+	});
+
+	Object.keys(patch).forEach(function (key) {
+		if (patch[key] === null) {
+			console.log(`Deleting config var: ${key}`); // eslint-disable-line no-console
+		} else if (patch[key] !== current[key]) {
+			console.log(`Setting config var: ${key}`); // eslint-disable-line no-console
+		}
+	});
+
+	console.log('Setting config vars', Object.keys(patch)); // eslint-disable-line no-console
```

Comment:
Is this duplicating the logging above? 🤔

Reply:
I thought it's ok as we log what is happening with:
```
Retrieving pipeline details from Heroku...
Retrieving current and desired config vars...
```
and then this signals that they are being set and it prints out every var that it is setting.